### ISSUE TITLE
Cannot build sys image- Win 64 0.5.1

### ISSUE THREAD

**Issue by StephenVavasis** [JuliaLang/julia#20958]
I am trying to patch an earlier bug I reported on this website (regarding arrays with 6 or more subscripts) in 0.5.1.  However, the instructions for building a system image from the documentation are apparently not working for me.  This is Julia 0.5.1 for Windows 10 - 64 bit that I installed yesterday via the self-installer from the Julia download site.  I issued the commands:
```julia
julia> include(joinpath(JULIA_HOME, Base.DATAROOTDIR, "julia", "build_sysimg.jl"))
```
and
```
julia> build_sysimg(force=true)
```
After many lines flew by on my console window, eventually I received a message stating that the build was successful:
```
C:\Users\vavasis\AppData\Local\Julia-0.5.1\share\julia\base\precompile.jl
INFO: System image successfully built at C:\Users\vavasis\AppData\Local\Julia-0.5.1\bin\..\lib\julia\sys.ji
INFO: Julia will automatically load this system image at next startup
```

However, the old sys.dll was not overwritten as you can see from the date:

```
c:\Users\vavasis\AppData\Local\Julia-0.5.1\lib\julia>dir
 Volume in drive C is Windows8_OS
 Volume Serial Number is 88DB-A798

 Directory of c:\Users\vavasis\AppData\Local\Julia-0.5.1\lib\julia

2017-03-08  11:54 PM    <DIR>          .
2017-03-08  11:54 PM    <DIR>          ..
2017-03-08  11:52 PM         2,116,720 inference.ji
2017-03-08  11:52 PM         7,253,763 inference.o
2017-03-05  10:47 AM        42,798,819 sys-debug.dll
2017-03-05  10:37 AM        41,515,369 sys.dll
2017-03-08  11:54 PM        26,094,220 sys.ji
2017-03-08  11:54 PM        48,834,489 sys.o
               6 File(s)    168,613,380 bytes
               2 Dir(s)  26,042,474,496 bytes free
```

**Comment by tkelman** [comment]
> After many lines flew by on my console window

One of those was probably a warning that you need to install a toolchain in order to have a linker. It should say something to the effect of `Pkg.add("WinRPM"); using WinRPM; WinRPM.add("gcc")`

**Comment by StephenVavasis** [comment]
Indeed, this message did fly by.  I installed the necessary packages and was able to rebuild julia 0.5.1 so that 6-subscript arrays now work properly.

Here is the message:

```
WARNING: Install GCC via `Pkg.add("WinRPM"); WinRPM.install("gcc")` to generate sys.dll for faster startup times
WARNING: No supported compiler found; startup times will be longer
```

So, as suggested, I fixed the problem with the installation of gcc.  But there is still an issue: this error message about the missing gcc package should be clearer and more visible.

**Comment by musm** [comment]
I'm pretty sure the underlying issue here has now been fixed

**Comment by tkelman** [comment]
Nothing was fixed here. "there is still an issue: this error message about the missing gcc package should be clearer and more visible." Suggestions welcome on how to accomplish that.

**Comment by musm** [comment]
Is the following warning not sufficient, seem pretty clear to me?
```
warn("Install GCC via `Pkg.add(\"WinRPM\"); WinRPM.install(\"gcc\")` to generate sys.dll for faster startup times.")
```

The error regarding possible failures during linking and so forth have been resolved

**Comment by StephenVavasis** [comment]
My problem was that the above message about GCC appears in the middle of a long stream of messages in the REPL and quickly flies off the screen, and then the final message is that the build was successful even though it was not.  I haven't checked to see if this is fixed in 0.5.2 or 0.6.0 because I haven't needed a custom system image in 0.5.2 or 0.6.0.  (For 0.5.1, I needed a custom system image to fix an unpatched bug.)

**Comment by musm** [comment]
I see, ok personally I find the color difference visually distinctive enough, but it may not be a bad idea to print the warnings after everything since the list of printed statements is rather large so I can see how things can be lost
